in-progress/8754-slashing-and staking.md

```diff
@@ -0,0 +1,357 @@
+
+# Proof Of Governance Slashing
+
+Why do we slash?
+
+We need to make sure the chain is always (eventually) finalizing new blocks. The conditions required for the chain to finalize new blocks are:
+1) Sequencers are making valid block proposals
+2) More than 2/3 of the committee is attesting on valid block proposals.
+3) Provers can obtain timely proof data for txs included in the epoch.
+4) Provers post proofs to the L1 on time.
+
+While Based Fallback acts as a backstop that ensures eventual liveness, malicious or permanently offline validators should eventually be kicked out of the validator set otherwise we end up in Based Fallback mode all too often. 
+
+In PoG, validators vote to slash any dishonest validators. To reduce coordination costs, we equip the validator client software with the ability to automatically initiate and respond to votes to slash dishonest validators based on onchain or offchain (i.e. p2p) evidence.
+
+Any validator can initiate a proposal to slash any other validator(s). A proposal to slash must include the offense, epoch/slot information and the validators to be slashed.  
+
+Validators vote by signing the slashing proposal and gossiping back the signed message over the p2p. Anyone can submit a slashing proposal that has gathered enough signatures to the L1 for verification. 
+
+Based on the offense in question, the rollup contract establishes max slashable stakes. The rollup contract also verifies certain conditions have occured before executing the slash. 
+
+This is more subjective than purely onchain slashing but it enables precision that is not possible with purely onchain slashing. We can adopt Ethereum's principle of "honest validators should not be slashed" at the expense of increased coordination cost.
+
+
+### Users of PoG Slashing
+
+Validators are the users of the PoG slashing mechanism. Full nodes can contribute data to validators but they should not be able to vote for a slashing proposal. 
+
+
+## PoG Slashable Offences
+
+These are actions that can and will be slashed using PoG.
+
+### 1. Committee signs off on an invalid state root
+
+Some validators implement a `ALWAYS_EXECUTE` flag which causes them to execute every block regardless of whether they're in the committee or not. Validators who implement this flag are called "Executing Validators". 
+
+After every succesful block proposal, these Executing Validators will retrieve the list of `TxHash` from L1 and attempt to re-execute all transactions in the block. 
+
+In the happy path, the Executing Validators are able to obtain the `TxObjects` and proofs required to do so. If the resulting state root does not match what was published to L1, they read the list of all committee members who signed the invalid block. 
+
+The Executing Validator prepares a payload containing:
+1) The offense in question: Invalid state root
+2) The slot number of the invalid block.
+3) The list of committee members who signed off on the invalid block.
```

Comment:
Should a validator accept this proposal if only a subset of mis-signing committee members are included in the proposal, this seems like a decent amount of work here. 

Say I controlled some of the nodes who may get slashed for this, since creating a proposal here seems free, I would spam proposals that include my other committe members getting slashed and not my own node. ?

Also in the case we cover this by requiring validators to check all signatures that mis signed, each of these proposals makes nodes do a more work, so they should ideally have a cost associated with them.